os/net/mac/tsch/sixtop/sixp.c

```diff
@@ -351,13 +416,16 @@ sixp_output(sixp_pkt_type_t type, sixp_pkt_code_t code, uint8_t sfid,
   nbr = sixp_nbr_find(dest_addr);
 
   /*
-   * Make sure we have a nbr for the peer if the packet is a response with
-   * success so that we can manage the schedule generation.
+   * Make sure we have a nbr for the peer if the packet is a response
+   * with success so that we can manage the schedule generation unless
+   * the request is CLEAR or in an unsupported format, in such a case
+   * we're going to return RC_ERR_VERSION or RC_ERR_SFID.
    */
   if(nbr == NULL &&
-     type == SIXP_PKT_TYPE_RESPONSE && code.value == SIXP_PKT_RC_SUCCESS &&
-     ((cmd = sixp_trans_get_cmd(trans)) == SIXP_PKT_CMD_ADD ||
-      cmd == SIXP_PKT_CMD_DELETE) &&
+     type == SIXP_PKT_TYPE_RESPONSE &&
+     sixp_trans_get_cmd(trans) != SIXP_PKT_CMD_CLEAR &&
+     (code.value != SIXP_PKT_RC_ERR_VERSION ||
+      code.value != SIXP_PKT_RC_ERR_SFID) &&
      (nbr = sixp_nbr_alloc(dest_addr)) == NULL) {
     LOG_ERR("6P: sixp_output() fails because of no memory for another nbr\n");
```

Comment:
The condition `(code.value != SIXP_PKT_RC_ERR_VERSION || code.value != SIXP_PKT_RC_ERR_SFID)` is equal to `!(code.value == SIXP_PKT_RC_ERR_VERSION && code.value == SIXP_PKT_RC_ERR_SFID)`, which is always true. It's a mistake, right? 

I guess the correct condition is:

```c
nbr == NULL &&
type == SIXP_PKT_TYPE_RESPONSE &&
sixp_trans_get_cmd(trans) != SIXP_PKT_CMD_CLEAR &&
code.value != SIXP_PKT_RC_ERR_VERSION &&
code.value != SIXP_PKT_RC_ERR_SFID &&
(nbr = sixp_nbr_alloc(dest_addr)) == NULL
```

Reply:
Great catch! Fixing it.